File: API/apollo_api_services/src/main/java/com/apollo/entity/Address.java
package com.apollo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "address")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String district;
    private String ward;
    private String city;
    private String street;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "shipper_id")
    private Shipper shipper;

    @ManyToOne
    @JoinColumn(name = "admin_id")
    private Admin admin;

    @OneToMany(mappedBy = "address")
    @JsonIgnore
    private Set<ShopOrder> shopOrders;
}

File: Mobile App/Apollo Deliver/lib/Models/Address.dart
import 'package:apollodeliver/Models/Admin.dart';
import 'package:apollodeliver/Models/Shipper.dart';
import 'package:apollodeliver/Models/ShopOrder.dart';
import 'package:apollodeliver/Models/User.dart';

class Address {
  int? id;
  String? district;
  String? ward;
  String? city;
  String? street;
  User? user;
  Shipper? shipper;
  Admin? admin;
  Set<ShopOrder>? shopOrders;

  Address({
    this.id,
    this.district,
    this.ward,
    this.city,
    this.street,
    this.user,
    this.shipper,
    this.admin,
    this.shopOrders,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'],
      district: json['district'],
      ward: json['ward'],
      city: json['city'],
      street: json['street'],
      user: json['user'] != null ? User.fromJson(json['user']) : null,
      shipper: json['shipper'] != null ? Shipper.fromJson(json['shipper']) : null,
      admin: json['admin'] != null ? Admin.fromJson(json['admin']) : null,
      shopOrders: (json['shopOrders'] as List?)
          ?.map((item) => ShopOrder.fromJson(item))
          .toSet(),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'district': district,
    'ward': ward,
    'city': city,
    'street': street,
    'user': user?.toJson(),
    'shipper': shipper?.toJson(),
    'admin': admin?.toJson(),
    'shopOrders': shopOrders?.map((item) => item.toJson()).toList(),
  };
}
